### ISSUE TITLE
Print Dialog restrictions under wrong permission 

### ISSUE THREAD

**Issue by MordyT** [sumatrapdfreader/sumatrapdf#4739]
Hello,
Long time Sumatra user. I took a quick look through open issues and discussions and don’t see this mentioned. 

We use the sumatrapdfrestrict.ini function to control the user experience (see https://github.com/sumatrapdfreader/sumatrapdf/blob/master/docs/sumatrapdfrestrict.ini) and ran into an interesting situation.

Per documentation above, if DiskAccess is turned off, the print dialog is not shown. This is true in every version of Sumatra 2.5 and above. In 2.4 (and below), it is shown, while leaving the other functions controlled by DiskAccess disabled.

We’d like to allow the print dialog without the rest of the features controlled by DiskAccess, but we’d also like to be able to upgrade from 2.4. Is it possible to move the print dialog into the PrinterAccess function instead of DiskAccess or as a separate permission? Makes more sense that PrinterAccess would control printing features anyways.

**Comment by GitHubRulesOK** [comment]
Hmmm 
not the developer and time changes some security facets / methods  but the reason for that dialog being under no disk access is there are loopholes in that area so for example Print to PDF gives you full ability to access the root of the drives and upwards.

**Comment by MordyT** [comment]
Hmm, that's a fair point (the print to PDF opening the door for file system access). However, it's not mitigated because I can just set my PDF printer as default and get the same dialog.

It could be that I'm stuck in an XY Problem. Is there another way to allow the user to select which pages to print besides that dialog in Sumatra that I just don't know?

**Comment by GitHubRulesOK** [comment]
SumatraPDF is based upon MuPDF VIEW which does not have any of Artifex's printer code (GhostScript)

Thus "Printing" has been a weak feature of SumatraPDF for a long time.  GhostScript print a page or pages would be better.
You could write a "shim" to pass pages numbers to a set of GhostScript calls from SumatraPDF but would be simpler not to.

Ghostscript as a console application does not use a print dialog so only expects one range per call "first page to last page"
Thus the easiest way to get prints is write a dialog for page numbers to pass to GS direct.

**Comment by kjk** [comment]
makes sense, I removed print dialog restriction when no disk access but please test because I didn't
